main.go

```diff
@@ -135,14 +137,17 @@ func main() {
 	// +kubebuilder:scaffold:builder
 
 	ctx := ctrl.SetupSignalHandler()
+	store, err := memorystore.New(&memorystore.Config{
+		Interval: 10 * time.Minute,
+	})
 
 	setupLog.Info("starting event server", "addr", eventsAddr)
 	eventServer := server.NewEventServer(eventsAddr, log, mgr.GetClient())
-	go eventServer.ListenAndServe(ctx.Done())
+	go eventServer.ListenAndServe(ctx.Done(), store)
```

Comment:
Either the error should be handled or the function should not return an error.

Reply:
What error are we taking about? If the server listener encounters an error it crashes the whole process https://github.com/fluxcd/notification-controller/blob/main/internal/server/receiver_server.go#L59

Reply:
This is a new error introduced when creating the middleware, but I will solve this by also exiting.